Pod/Illuminator/IlluminatorTestProgress.swift

```diff
@@ -9,6 +9,42 @@
 import Foundation
 import XCTest
 
+/*
+ * This protocol allows custom handling of results.  For example, screenshots of failure states or desktop notifications
+ * isPass and isFail can both be false -- it indicates a "Flagging" state.  They are guaranteed to not both be true
+ */
+public protocol IlluminatorTestResultHandler {
+    associatedtype AbstractStateType
```

Comment:
Unfortunately we do need the associatedType... it's the only way to get a generic protocol (see link a few lines down for https://milen.me/writings/swift-generic-protocols/